src/software/networking/proto_udp_listener.hpp

```diff
@@ -153,14 +171,42 @@ void ProtoUdpListener<ReceiveProtoT>::startListen()
 template <class ReceiveProtoT>
 void ProtoUdpListener<ReceiveProtoT>::handleDataReception(
     const boost::system::error_code& error, size_t num_bytes_received)
+{
+    std::mutex m;
+    std::condition_variable cv;
+
+    std::thread recv_thread([this, &cv, &error, &num_bytes_received]() {
+        receiveData(error, num_bytes_received);
+        cv.notify_one();
+    });
+
+    recv_thread.detach();
```

Comment:
Could you document the logic here? I'm a bit lost on what's happening. It seems like the mutex isn't shared between function calls, so will it ever block? Also, why is a condition_variable needed?

Reply:
the condition variable is associated with the mutex, basically the issue that this code is trying to address is that during python's teardown, a c++ pointer into a python callback suddenly goes out of scope and hangs indefinitely

but it's good to document this more

Reply:
fixed